### DISCUSSION TITLE
Atomic rollouts

### DISCUSSION THREAD

**Discussion by aluzzardi** [dagger/dagger#2083] [Ideas]
When deploying `A` and `B`, if `A` succeeds then `B` fails the rollout is left as is. This can result in a broken deployment since we now have a new version of `A` with an old version of `B`.

Ideally, if a version N of a plan fails, dagger should be re-apply N-1 to bring back `A` and `B` to the same version.

/cc @samalba

**Comment by samalba** [comment]
cc @gvidon as it was his request during the last demo.
